core/templates/dev/head/i18n.js

```diff
@@ -75,8 +75,9 @@ oppia.config([
   var availableLanguageKeys = [];
   var availableLanguageKeysMap = {};
   for (var prop in GLOBALS.SUPPORTED_SITE_LANGUAGES) {
-    availableLanguageKeys.push(prop);
-    availableLanguageKeysMap[prop + '*'] = prop;
+    availableLanguageKeys.push(GLOBALS.SUPPORTED_SITE_LANGUAGES[prop].id);
```

Comment:
So, this actually turned into a bit of a rabbit hole, but in the process I found some issues with our existing karma tests -- GLOBALS was being overwritten in some tests, and this was causing the default values to not get picked up.

I've fixed these problems in #2443. Once that's merged in, you should be able to bring that into your branch, modify the constant in core/tests/karma-globals.js accordingly, and this should work! Thanks for bringing this up :)